modules/auth-allowing-javascript-access-api-server.adoc

```diff
@@ -0,0 +1,63 @@
+// Module included in the following assemblies:
+//
+// * authentication/configuring-corsallowedorigins.adoc
+
+[id="auth-allowing-javascript-access-api-server_{context}"]
+= Allowing JavaScript-based access to the API server from additional hosts
+
+By default, localhost and 127.0.0.1 are whitelisted to access the API server and
+OAuth server safely even when the browser would consider the requests to be
+link:http://www.w3.org/TR/cors/[cross-origin].
+
+If you need to access the API server or OAuth server from a JavaScript
+application using a different host name, you can configure additional host names
+to allow.
+
+.Prerequisites
+
+* Access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Edit the API servers resource.
++
+----
+$ oc edit apiservers.config.openshift.io
+----
++
+. Add the `additionalCORSAllowedOrigins` field under the `spec` section and
+specify one or more additional host names.
++
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: APIServer
+metadata:
+  annotations:
+    release.openshift.io/create-only: "true"
+  creationTimestamp: "2019-07-11T17:35:37Z"
+  generation: 1
+  name: cluster
+  resourceVersion: "907"
+  selfLink: /apis/config.openshift.io/v1/apiservers/cluster
+  uid: 4b45a8dd-a402-11e9-91ec-0219944e0696
+spec:
+  additionalCORSAllowedOrigins:
+  - (?i)//my\.subdomain\.domain\.com(:|\z) <1>
+----
+<1> The host name is specified as a Golang regular expression that matches
```

Comment:
Should we include the Golang regexp documentation link? https://github.com/google/re2/wiki/Syntax

Reply:
Unfortunately we've been asked not to link to GitHub repos in the product docs. Unless @sferich888 wants to make an exception here.

Reply:
This link here is OK, as it's referencing an uncommon regex syntax (would be good to get a link version locked to what we build with). Or better if we could pull over the necessary docs for the syntax parser.

Reply:
Great thanks @sferich888!

FYI @stlaz I've added this link now. Let me know if there's a particular version of this wiki content that you think we should link to, that matches the version that OpenShift currently supports.

Reply:
I think we should be ok with the page here. The set of regular expressions of a language does not change that often (if ever), I think we should be safe with whatever appears on the upstream wiki.